### DISCUSSION TITLE
How do I connect the iOS Audio Player Controls?

### DISCUSSION THREAD

**Discussion by joshualambert** [appcelerator/titanium_mobile#12656] [Q&A]
Hey guys! I'm using background audio with the following code:

```
player.streamer = Ti.Media.createAudioPlayer({
  allowBackground: true,
  audioFocus: true
});

player.streamer.url = "AUDIO URL HERE";

player.streamer.start();

```

![IMG_6548](https://user-images.githubusercontent.com/1421842/112388863-cb297280-8cc1-11eb-9511-43fd1c57e4e4.jpg)

The code itself works fine for playing the remote MP3 file, and survives the app going into the background. However the audio controls in the app drawer appear to have no effect on the audio playing. Play/pause/fast-forward/rewind do not work. Also don't see a way in the documentation to set cover-art. Is there something I'm missing in the Ti documentation somewhere? Has someone done this before, perhaps using Hyperloop? If so, can you share some code examples? THANKS!!! :)

**Comment by Topener** [comment]
I think you need to use the [musicplayer](https://titaniumsdk.com/api/titanium/media/musicplayer.html) for that.

**Comment by macCesar** [comment]
Did you check this?

https://github.com/hoyo/TiRemoteControl

It is old.. but you can analize how it works!!!
